### DISCUSSION TITLE
next/image - Add threshold for image lazy loading

### DISCUSSION THREAD

**Discussion by wasddd431** [vercel/next.js#22681] [Ideas]
**Describe the feature you'd like to request**

For lazy loading, we'd like to set a certain `threshold` in order to load images earlier than the default behavior.

For some of our users - especially mobile - the images are loaded "too late" and since we're not yet using a placeholder (waiting for #18858 to be implemented), it sometimes looks a bit "flashy" even though there's no CML. But still, the UX could improve by fetching images earlier.

**Describe the solution you'd like**

By having a `threshold` property for the `<Image />` component provided by `next/image`, the developers could dynamically choose when to load the images instead of relying on the default behavior given by the package / native browser loading behavior. The property might accept the `threshold` as `number`, so it might look like this:

```javascript
<Image 
  src={src}
  alt={alt}
  width={width}
  height={height}
  threshold={20}
/>
```

The `threshold` might be understood as _percentage until the image hits the viewport_.

**Describe alternatives you've considered**

I considered using https://github.com/cyrilwanner/next-optimized-images but I don't really see the point on switching the image flow since I'm genuinely happy with the `next/image` package - I'd simply like to have _a bit_  more accessibility - especially when it comes to the `how` and `when` to fetch the images.

**PS: I searched the issues for a similar request, but could not find any, so I am sorry in case this indeed is a duplicate of another issue.**

**Comment by AlexandraKlein** [comment]
Any update on this? My lazy loaded images are also loading very late/slowly, especially when using `layout="responsive"` and `layout="fill"`.

**Reply by wasddd431** [reply]
@AlexandraKlein unfortunately not - however upgrading to the latest next.js version helped me a lot. And we also found a difference between dev + prod.
